File: sdk/nowyouseeme/operation_types.py
"""
Operation types - these are the ONLY valid operation types
Must match backend models/operation_types.go
"""

from enum import Enum


class OperationType(str, Enum):
    """Operation type enumeration"""

    # Goal operations
    GOAL_CREATE = "goal_create"
    GOAL_TRANSITION = "goal_transition"
    GOAL_UPDATE = "goal_update"
    GOAL_COMPLETE = "goal_complete"
    GOAL_ABANDON = "goal_abandon"

    # Capability operations
    CAPABILITY_ADD = "capability_add"
    CAPABILITY_REMOVE = "capability_remove"
    CAPABILITY_UPDATE = "capability_update"

    # Limitation operations
    LIMITATION_ADD = "limitation_add"
    LIMITATION_REMOVE = "limitation_remove"
    LIMITATION_UPDATE = "limitation_update"

    # Aspiration operations
    ASPIRATION_ADD = "aspiration_add"
    ASPIRATION_REMOVE = "aspiration_remove"
    ASPIRATION_UPDATE = "aspiration_update"

    # Metadata operations
    METADATA_UPDATE = "metadata_update"


def get_all_operation_types():
    """Get all valid operation types"""
    return [op.value for op in OperationType]


def is_valid_operation_type(value: str) -> bool:
    """Check if a string is a valid operation type"""
    try:
        OperationType(value)
        return True
    except ValueError:
        return False

File: frontend/src/api/operation-types.ts
/**
 * Operation types - these are the ONLY valid operation types
 * Must match backend models/operation_types.go
 */

export const OperationType = {
  // Goal operations
  GoalCreate: "goal_create",
  GoalTransition: "goal_transition",
  GoalUpdate: "goal_update",
  GoalComplete: "goal_complete",
  GoalAbandon: "goal_abandon",

  // Capability operations
  CapabilityAdd: "capability_add",
  CapabilityRemove: "capability_remove",
  CapabilityUpdate: "capability_update",

  // Limitation operations
  LimitationAdd: "limitation_add",
  LimitationRemove: "limitation_remove",
  LimitationUpdate: "limitation_update",

  // Aspiration operations
  AspirationAdd: "aspiration_add",
  AspirationRemove: "aspiration_remove",
  AspirationUpdate: "aspiration_update",

  // Metadata operations
  MetadataUpdate: "metadata_update",
} as const

export type OperationType = typeof OperationType[keyof typeof OperationType]

/**
 * Get all valid operation types
 */
export function getAllOperationTypes(): OperationType[] {
  return Object.values(OperationType)
}

/**
 * Check if a string is a valid operation type
 */
export function isValidOperationType(value: string): value is OperationType {
  return Object.values(OperationType).includes(value as OperationType)
}
